testsuite/python/domain_decomposition.py

```diff
@@ -63,6 +63,14 @@ def test_min_num_cells(self):
         self.assertGreaterEqual(n_cells, cs.min_num_cells)
         self.assertLessEqual(n_cells, cs.max_num_cells)
 
+    def test_position_rounding(self):
+        """This places a particle on the box boundary,
+           with parameteres that could cause problems with
+           rounding."""
+        self.system.min_global_cut = 12.0 / 4.25
+        self.system.part.add(pos=[25, 25, 0])  
+        self.assertEqual(1, len(self.system.part))
```

Comment:
I didn't observe particle duplication when running the script with multiple MPI threads. Wouldn't particle duplication cause a segfault due to a double call of `free()` on the same particle pointer?

Also, this test will silently stop working if someone changes the `box_l` at the top of the class. Should we re-define `system.box_l` within the body of that method to prevent that from happening?

Reply:
< I didn't observe particle duplication when running the script with multiple MPI threads. Wouldn't particle duplication cause a segfault due to a double call of free() on the same particle pointer?

No, the particles is created on multiple nodes, not multiple times on the same node.

Reply:
> No, the particle is created on multiple nodes, not multiple times on the same node.

`print(len(system.part))` prints 1 on my machine with multiple MPI threads and without your fix, so what is the assertEqual guarding against?